src/hotspot/share/utilities/vmError.cpp

```diff
@@ -1690,6 +1726,8 @@ void VMError::report_and_die(int id, const char* message, const char* detail_fmt
     // if os::abort() doesn't abort, try os::die();
     os::die();
   }
+  // We REALLY better not reach here.
+  ShouldNotReachHere();
```

Comment:
Maybe just `os::die()` again rather than a recursive call back into VMError code? (Not that we ever expect to execute this but ...

Reply:
This blob of code could have been equivalently written as
```
  switch (after_report_action) {
  case AfterReportAction::Return:
    return;
  case AfterReportAction::Die:
    ... breakpoint and abort ...
    os::die();
    // should be unreachable because die is noreturn.  perhaps fall through
    // if get here, though we're in UB territory.
  default:
    ShouldNotReachHere();
  }
```
I didn't write it that way because the Die code is a bit longer and more
complicated than I like to put in a case clause.